.circleci/config.yml

```diff
@@ -9,29 +9,33 @@ references:
   restore-dependencies-cache: &restore-dependencies-cache
     restore_cache:
       keys:
-        - deps-py<< parameters.version >>-{{ checksum "poetry.lock" }}
+        - deps-py<< parameters.python-version >>-django<< parameters.django-version >>-{{ checksum "poetry.lock" }}
   install-dependencies: &install-dependencies
     run:
       name: Install Dependencies
       command: |
         source $HOME/.poetry/env
         poetry install
+        poetry run pip install "django>=<< parameters.django-version >>"
   save-dependencies-cache: &save-dependencies-cache
     save_cache:
-      key: deps-py<< parameters.version >>-{{ checksum "poetry.lock" }}
+      key: deps-py<< parameters.python-version >>-django<< parameters.django-version >>-{{ checksum "poetry.lock" }}
       paths:
         - /home/circleci/.cache/pypoetry/virtualenvs
   parametrised-python-executor: &parametrised-python-executor
     parameters:
-      version:
+      python-version:
+        type: string
+      django-version:
         type: string
     executor:
       name: python
-      version: << parameters.version >>
-  python-version-matrix: &python-version-matrix
+      version: << parameters.python-version >>
+  version-matrix: &version-matrix
     matrix:
       parameters:
-        version: ["3.5.5", "3.6", "3.7", "3.8"]
+        python-version: ["3.6", "3.7", "3.8", "3.9"]
+        django-version: ["2.2", "3.1"]
```

Comment:
Just checking it's intentional not to test on Django 3.0?

Reply:
Not really, mostly matching the behaviour as we're doing 2.2 but not 2.1 and 2.0.

Reply:
Oh, I'd assumed that we weren't supporting those versions. If they're still supported I think we ought to be testing on them. (though maybe not linting on all the intermediate versions?)

Reply:
I've now refreshed on Django's lifecycle, 2.2 is last LTS so yeah I'll add 3.0.

Reply:
Hrm, my reading of https://www.djangoproject.com/download/#unsupported-versions says that 3.0 is in fact already unsupported, which I hadn't realised. Do we want to exclude that? It may not be worth it though?

Reply:
I think this is recent enough that unless it takes effort not too we may as well do it.